src/ucp/core/ucp_am.c

```diff
@@ -1382,17 +1383,24 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_am_long_first_handler,
      * Need to allocate a separate rdesc which would be in one contigious chunk
      * with data buffer. */
     first_rdesc = ucs_malloc(first_hdr->total_size + sizeof(ucp_recv_desc_t) +
-                             sizeof(*first_hdr),
+                                     sizeof(*first_hdr) + user_hdr_length,
```

Comment:
duplicate of 1362-1663?

Reply:
no, on 1362 i calculate the data length in this message fragment.
Here I calculate the amount of memory needed for: the whole message, rdesc, user hdr, AM first hdr

Reply:
in both places, there is a calc of first_hdr->total_size + sizeof(*first_hdr) + user_hdr_length

Reply:
according to the calculation, seems can do like this:
```
x         = first_hdr->total_size + sizeof(*first_hdr) + user_hdr_length
remaining = x - am_length

...
ucs_malloc(x + sizeof(ucp_recv_desc_t))
```